rsec16/coder.go

```diff
@@ -175,6 +175,12 @@ func makeReconstructionMatrix(dataShards int, availableRows, missingRows, usedPa
 	return m.RowReduceForInverse(n)
 }
 
+type IrrepairableError struct{}
+
+func (IrrepairableError) Error() string {
+	return "repair necessary but not possible"
+}
+
 // ReconstructData takes a list of data shards and parity shards, some
 // of which may be nil, and tries to reconstruct the missing data
 // shards. If successful, the nil rows of data are filled in and a nil
```

Comment:
Can you add in the comment that `NotEnoughParityShardsError` (what I suggested `IrrepairableError` be renamed to in another comment) is one of the possible errors that can be returned? Something like:

> Otherwise, an error is returned. In particular, if there are missing data shards but there aren't enough parity shards to reconstruct them, `NotEnoughParityShardsError` is returned.